scene/resources/packed_scene.cpp

```diff
@@ -539,9 +573,38 @@ Node *SceneState::instantiate(GenEditState p_edit_state) const {
 			Dictionary paths = dnp.value;
 
 			bool valid;
-			Dictionary dict = dnp.base->get(dnp.property, &valid);
+			Variant property_value = dnp.base->get(dnp.property, &valid);
 			ERR_CONTINUE_EDMSG(!valid, vformat("Failed to get property '%s' from node '%s'.", dnp.property, dnp.base->get_name()));
-			dict = dict.duplicate();
+
+			Dictionary dict;
+			if (property_value.get_type() != Variant::DICTIONARY) {
+				// The value of the property is currently not a dictionary.
+				// This can happen if we have an exported dictionary from .NET with its default value of null.
+				// This means we have to find the property in the property list and parse out the key and value types.
+
+				List<PropertyInfo> all_dnp_base_props;
+				dnp.base->get_property_list(&all_dnp_base_props);
+				for (const PropertyInfo &dnp_base_prop : all_dnp_base_props) {
+					if (dnp_base_prop.name == dnp.property) {
+						if (dnp_base_prop.type == Variant::DICTIONARY) {
+							Variant::Type key_subtype;
+							PropertyHint key_subtype_hint;
+							String key_class_name;
+							Variant::Type value_subtype;
+							PropertyHint value_subtype_hint;
+							String value_class_name;
+
+							_parse_dictionary_hint_string(dnp_base_prop.hint_string, key_subtype, key_subtype_hint, &key_class_name, value_subtype, value_subtype_hint, &value_class_name);
+
+							dict.set_typed(key_subtype, StringName(key_class_name), Variant(), value_subtype, StringName(value_class_name), Variant());
```

Comment:
This is not entirely correct if the type arguments for the dictionary are script types. However, the hint string does not include any information about scripts, so I'm not sure what the best way to address it is.